### ISSUE TITLE
 Unable to collect wrapped bales 1.0.3.4

### ISSUE THREAD

**Issue by FarmerTreanor** [GMNGjoy/FS25_ContractBoost#34]
We seem have have the bug where the UAL was picking up the straw bales fine and now has stopped also tested using the in game Anderson RBM2000 and this provided the error you don't own the bale, using 1.0.3.4 of this mod and what logs do you need as its on MP so if server logs i can grap them and copy and paste them here.

**Comment by GMNGjoy** [comment]
In this case - what is the `enableCollectingBalesFromMissions` setting set to? Is that feature on or off? 

When you are trying to collect the bales and you walk up to them in game, does it show that you (or your farm) are the owner of the bales?

The logs that are needed would either be the server logs, or single player logs - which would be even better if you are running the mod with `debugMode="true"`

**Comment by FarmerTreanor** [comment]
First answer when walking up to bales showing no owners like would if bales it myself would show Example Own By : Farm, having looked into settings all turn on.

Log is included and added the xml here but as text file.
[log.txt](https://github.com/user-attachments/files/18202644/log.txt)
[contract.txt](https://github.com/user-attachments/files/18202647/contract.txt)

**Comment by GMNGjoy** [comment]
thank you for the logs, and the explanation. It looks like you are running / hosting a local MP server on your windows machine (not running it from a dedicated server), is that correct?

Your logs call out the following setting:
> `    enableCollectingBalesFromMissions :: false`
which means that you are _not allowed_ to collect bales from missions, which means that the mod is working correctly based on that setting (as intended).

It's also unclear if the correct configuration is being loaded as it appears that the mod is loading your _local_ settings file (which would only change those values for you), which in some way may cause the discrepency between what you "see" and what you are being actually granted,

> `2024-12-19 18:21   Info: ContractBoost:LOAD :: MODSETTINGS configuration from: C:/Users/petet/Documents/My Games/FarmingSimulator2025/modSettings/ContractBoost.xml | debug: false`

That said, the parent issue here may still be #3 or #30 - in that sometimes the server does not accurately reflect the settings set in the config file, or reverts to default settings (which has that setting turned off). I'm working on those issues, but need a dedicated server to do so; i'm waiting for it to spin up.

If you have access to the server, you should be able to either change that setting in-game (through the settings menu, don't forget to save) and / or edit the config file directly via FTP. both changes will require a restart. when you pull up the logs, look for this block:

```
2024-12-19 18:21   Info: ContractBoost:LOAD :: MODSETTINGS configuration from: C:/Users/petet/Documents/My Games/FarmingSimulator2025/modSettings/ContractBoost.xml | debug: false
2024-12-19 18:21   Info: ContractBoost:LOAD :: config loaded
2024-12-19 18:21   Info:    customMaxPerType: 
2024-12-19 18:21   Info:    customRewards: 
2024-12-19 18:21   Info:       baleMission :: 3500
2024-12-19 18:21   Info:       baleWrapMission :: 500
2024-12-19 18:21   Info:    debugMode :: false
2024-12-19 18:21   Info:    enableCollectingBalesFromMissions :: false
2024-12-19 18:21   Info:    enableContractValueOverrides :: true
2024-12-19 18:21   Info:    enableFieldworkToolFillItems :: true
2024-12-19 18:21   Info:    enableGrassFromMowingMissions :: true
2024-12-19 18:21   Info:    enableHayFromTedderMissions :: true
2024-12-19 18:21   Info:    enableStonePickingFromMissions :: true
2024-12-19 18:21   Info:    enableStrawFromHarvestMissions :: true
2024-12-19 18:21   Info:    enableSwathingForHarvestMissions :: true
2024-12-19 18:21   Info:    maxContractsOverall :: 50
2024-12-19 18:21   Info:    maxContractsPerFarm :: 10
2024-12-19 18:21   Info:    maxContractsPerType :: 5
2024-12-19 18:21   Info:    rewardFactor :: 1.5
2024-12-19 18:21   Info: ContractBoost :: loaded. debug: off
```

with that data you can verify very quickly if the setting is not only getting saved properly, but getting loaded from the correct settings file. in this case, it appears that there isn't a config file on the server, but it's defaulting to your local settings file, which may be part of the issue. I'll keep working on it, and let you know if I find anything!

**Comment by FarmerTreanor** [comment]
Ok done deep dive on the other server same issue and its because of this

<?xml version="1.0" encoding="utf-8" standalone="no"?>
<contractBoost>
    <settings>
        <debugMode>true</debugMode>
        <enableContractValueOverrides>true</enableContractValueOverrides>
        <enableStrawFromHarvestMissions>true</enableStrawFromHarvestMissions>
        <enableSwathingForHarvestMissions>true</enableSwathingForHarvestMissions>
        <enableGrassFromMowingMissions>true</enableGrassFromMowingMissions>
        <enableHayFromTedderMissions>true</enableHayFromTedderMissions>
        <enableStonePickingFromMissions>true</enableStonePickingFromMissions>
        <enableFieldworkToolFillItems>true</enableFieldworkToolFillItems>
        **<enableCollectingBalesFromMissions>false</enableCollectingBalesFromMissions>**
        <rewardFactor>0.500000</rewardFactor>
        <maxContractsPerFarm>10</maxContractsPerFarm>
        <maxContractsPerType>5</maxContractsPerType>
        <maxContractsOverall>50</maxContractsOverall>
    </settings>
    <customRewards>
        <baleMission>3500</baleMission>
        <baleWrapMission>500</baleWrapMission>
        <treeTransportMission>400</treeTransportMission>
    </customRewards>
</contractBoost>

This from other server same issue and believe because not applying to savegame FS25_ContractBoost.xml 

![FarmingSimulator2025Game_hd0ftfmISN](https://github.com/user-attachments/assets/fdc27f3d-545d-46f6-950c-0f5466805fbf)

**Comment by GMNGjoy** [comment]
That's what I've been hearing, thank you for the additional data.

**Comment by FarmerTreanor** [comment]
Just small update i edited the save game info so currently now saying true, this allow the bales to be pickup but the ownership of the bales still has nothing, this means can't sell or load them as bales into hay barn that i have takes straw bales.

**Comment by GMNGjoy** [comment]
> Just small update i edited the save game info so currently now saying true, this allow the bales to be pickup but the ownership of the bales still has nothing, this means can't sell or load them as bales into hay barn that i have takes straw bales.

just to confirm - this is still in MP that it's not working as expected?

**Comment by FarmerTreanor** [comment]
> > Just small update i edited the save game info so currently now saying true, this allow the bales to be pickup but the ownership of the bales still has nothing, this means can't sell or load them as bales into hay barn that i have takes straw bales.
> 
> just to confirm - this is still in MP that it's not working as expected?

correct for MP.

**Comment by FirenzeIT** [comment]
Hi @GMNGjoy, i have also various errors when i finished a contract (wrapped bales) and picked up the bales for selling them (before clicking finish on contract's section). Errors when i'm unloading the bales on animal's seller..

2024-12-23 22:36 Error: Can't change money of spectator farm
2024-12-23 22:36 LUA call stack:
  =dataS/scripts/FSBaseMission.lua:2720 addMoney
  =dataS/scripts/objects/SellingStation.lua:427 sellFillType
  =dataS/scripts/objects/SellingStation.lua:387 addFillLevelFromTool
  =dataS/scripts/triggers/UnloadTrigger.lua:229 addFillUnitFillLevel
  =dataS/scripts/triggers/BaleUnloadTrigger.lua:131 update
  =dataS/scripts/network/NetworkNode.lua:164 updateActiveObjects
  =dataS/scripts/network/Server.lua:64 update
  =dataS/scripts/BaseMission.lua:1017 update
  =dataS/scripts/FSBaseMission.lua:1936 update
  =dataS/scripts/mission00.lua:658 update
  =dataS/scripts/main.lua:1195 update
2024-12-23 22:36 Error: Can't change money of spectator farm
2024-12-23 22:36 LUA call stack:
  =dataS/scripts/FSBaseMission.lua:2720 addMoney
  =dataS/scripts/objects/SellingStation.lua:427 sellFillType
  =dataS/scripts/objects/SellingStation.lua:387 addFillLevelFromTool
  =dataS/scripts/triggers/UnloadTrigger.lua:229 addFillUnitFillLevel
  =dataS/scripts/triggers/BaleUnloadTrigger.lua:131 update
  =dataS/scripts/network/NetworkNode.lua:164 updateActiveObjects
  =dataS/scripts/network/Server.lua:64 update
  =dataS/scripts/BaseMission.lua:1017 update
  =dataS/scripts/FSBaseMission.lua:1936 update
  =dataS/scripts/mission00.lua:658 update
  =dataS/scripts/main.lua:1195 update
2024-12-23 22:37 Error: Can't change money of spectator farm
2024-12-23 22:37 LUA call stack:
  =dataS/scripts/FSBaseMission.lua:2720 addMoney
  =dataS/scripts/objects/SellingStation.lua:427 sellFillType
  =dataS/scripts/objects/SellingStation.lua:387 addFillLevelFromTool
  =dataS/scripts/triggers/UnloadTrigger.lua:229 addFillUnitFillLevel
  =dataS/scripts/triggers/BaleUnloadTrigger.lua:131 update
  =dataS/scripts/network/NetworkNode.lua:164 updateActiveObjects
  =dataS/scripts/network/Server.lua:64 update
  =dataS/scripts/BaseMission.lua:1017 update
  =dataS/scripts/FSBaseMission.lua:1936 update
  =dataS/scripts/mission00.lua:658 update
  =dataS/scripts/main.lua:1195 update
2024-12-23 22:37 Error: Can't change money of spectator farm

**Comment by FirenzeIT** [comment]
Add also a videoclip below:

[log#2.txt](https://github.com/user-attachments/files/18234276/log.2.txt)

[https://app.screencast.com/v8JUfQYSJHXIJ](url)

**Comment by FirenzeIT** [comment]
Hi, I tried it again, first collecting silage bales from the contracts (and storing them on my farm before collecting the reward and completing the mission), then after collecting the reward and completing the mission, loading them onto the lorry and taking them to the livestock dealer to sell them, always getting the errors below. So with this mod it is not possible to collect bales (especially silage) and then sell them, you have to use them only for animals or otherwise for your farm.

2024-12-25 00:41 Error: Can't change money of spectator farm
2024-12-25 00:41 LUA call stack:
  =dataS/scripts/FSBaseMission.lua:2720 addMoney
  =dataS/scripts/objects/SellingStation.lua:427 sellFillType
  =dataS/scripts/objects/SellingStation.lua:387 addFillLevelFromTool
  =dataS/scripts/triggers/UnloadTrigger.lua:229 addFillUnitFillLevel
  =dataS/scripts/triggers/BaleUnloadTrigger.lua:131 update
  =dataS/scripts/network/NetworkNode.lua:164 updateActiveObjects
  =dataS/scripts/network/Server.lua:64 update
  =dataS/scripts/BaseMission.lua:1017 update
  =dataS/scripts/FSBaseMission.lua:1936 update
  =dataS/scripts/mission00.lua:658 update
  =dataS/scripts/main.lua:1195 update

**Comment by FirenzeIT** [comment]
Hi @GMNGjoy, with the last update v1.3.0.49 you have fixed also errors from me mentioned above?

**Comment by GMNGjoy** [comment]
> above?

Not yet - there was no release posted as it's not working yet. MP support is first, then I'll address the bale issue. I'm getting close with the MP fix, and i have an idea on the reason the bales are having an issue. i've got a few days off work, I'm hoping to get a fix for both out asap. 

sorry for the wait, it's the holiday season and family takes priority.

**Comment by FirenzeIT** [comment]
Yes sure, 💯 i only reported the issues and ask why for this errors, take
your time 🙃

Il Gio 26 Dic 2024, 03:31 Gaming Joy | Josh ***@***.***> ha
scritto:

**Comment by FarmerTreanor** [comment]
> > above?
> 
> Not yet - there was no release posted as it's not working yet. MP support is first, then I'll address the bale issue. I'm getting close with the MP fix, and i have an idea on the reason the bales are having an issue. i've got a few days off work, I'm hoping to get a fix for both out asap.
> 
> sorry for the wait, it's the holiday season and family takes priority.

IRL always take first hope you have great Holiday Season with your family :)

**Comment by FirenzeIT** [comment]
Hi @GMNGjoy, now with last update 1.0.3.49 i can unload and sell bales of straw and silage without errors, but i have many errors, when i choose different option in settings... Before not having this type of errors...

2024-12-26 19:28   Info: FS25_ContractBoost:SETTINGS :: settingChanged table: 0x0000020c4723c088 0 customMaxPerType
2024-12-26 19:28 Error: Running LUA method 'update'.
C:/Users/User/Documents/My Games/FarmingSimulator2025/mods/FS25_ContractBoost/scripts/Settings.lua:36: attempt to index nil with table
2024-12-26 19:28   Info: FS25_ContractBoost:SETTINGS :: settingChanged table: 0x0000020c4723c088 1 customMaxPerType
2024-12-26 19:28 Error: Running LUA method 'update'.
C:/Users/User/Documents/My Games/FarmingSimulator2025/mods/FS25_ContractBoost/scripts/Settings.lua:36: attempt to index nil with table
2024-12-26 19:28   Info: FS25_ContractBoost:SETTINGS :: settingChanged table: 0x0000020c4723c088 2 customMaxPerType
2024-12-26 19:28 Error: Running LUA method 'update'.
C:/Users/User/Documents/My Games/FarmingSimulator2025/mods/FS25_ContractBoost/scripts/Settings.lua:36: attempt to index nil with table
2024-12-26 19:28   Info: FS25_ContractBoost:SETTINGS :: settingChanged table: 0x0000020c4723c088 3 customMaxPerType
2024-12-26 19:28 Error: Running LUA method 'update'.
C:/Users/User/Documents/My Games/FarmingSimulator2025/mods/FS25_ContractBoost/scripts/Settings.lua:36: attempt to index nil with table
2024-12-26 19:28   Info: FS25_ContractBoost:SETTINGS :: settingChanged table: 0x0000020c4723c088 4 customMaxPerType
2024-12-26 19:28 Error: Running LUA method 'update'.
C:/Users/User/Documents/My Games/FarmingSimulator2025/mods/FS25_ContractBoost/scripts/Settings.lua:36: attempt to index nil with table
2024-12-26 19:28   Info: FS25_ContractBoost:SETTINGS :: settingChanged table: 0x0000020c4723c088 5 customMaxPerType
2024-12-26 19:28 Error: Running LUA method 'update'.
C:/Users/User/Documents/My Games/FarmingSimulator2025/mods/FS25_ContractBoost/scripts/Settings.lua:36: attempt to index nil with table
2024-12-26 19:28   Info: FS25_ContractBoost:SETTINGS :: settingChanged table: 0x0000020c4723c088 6 customMaxPerType
2024-12-26 19:28 Error: Running LUA method 'update'.
C:/Users/User/Documents/My Games/FarmingSimulator2025/mods/FS25_ContractBoost/scripts/Settings.lua:36: attempt to index nil with table
2024-12-26 19:29   Info: FS25_ContractBoost:SETTINGS :: settingChanged table: 0x0000020c4723c088 7 customMaxPerType
2024-12-26 19:29 Error: Running LUA method 'update'.
C:/Users/User/Documents/My Games/FarmingSimulator2025/mods/FS25_ContractBoost/scripts/Settings.lua:36: attempt to index nil with table
2024-12-26 19:29   Info: FS25_ContractBoost:SETTINGS :: settingChanged table: 0x0000020c4723c088 8 customMaxPerType
2024-12-26 19:29 Error: Running LUA method 'update'.
C:/Users/User/Documents/My Games/FarmingSimulator2025/mods/FS25_ContractBoost/scripts/Settings.lua:36: attempt to index nil with table
2024-12-26 19:29   Info: FS25_ContractBoost:SETTINGS :: settingChanged table: 0x0000020c4723c088 9 customMaxPerType
2024-12-26 19:29 Error: Running LUA method 'update'.
C:/Users/User/Documents/My Games/FarmingSimulator2025/mods/FS25_ContractBoost/scripts/Settings.lua:36: attempt to index nil with table
2024-12-26 19:29   Info: FS25_ContractBoost:SETTINGS :: settingChanged table: 0x0000020c4723c088 10 customMaxPerType
2024-12-26 19:29 Error: Running LUA method 'update'.
C:/Users/User/Documents/My Games/FarmingSimulator2025/mods/FS25_ContractBoost/scripts/Settings.lua:36: attempt to index nil with table
2024-12-26 19:29   Info: FS25_ContractBoost:SETTINGS :: settingChanged table: 0x0000020c4723c088 11 customMaxPerType
2024-12-26 19:29 Error: Running LUA method 'update'.
C:/Users/User/Documents/My Games/FarmingSimulator2025/mods/FS25_ContractBoost/scripts/Settings.lua:36: attempt to index nil with table
2024-12-26 19:29   Info: FS25_ContractBoost:SETTINGS :: settingChanged table: 0x0000020c4723c088 12 customMaxPerType
2024-12-26 19:29 Error: Running LUA method 'update'.
C:/Users/User/Documents/My Games/FarmingSimulator2025/mods/FS25_ContractBoost/scripts/Settings.lua:36: attempt to index nil with table
2024-12-26 19:29   Info: FS25_ContractBoost:SETTINGS :: settingChanged table: 0x0000020c4723c088 13 customMaxPerType
2024-12-26 19:29 Error: Running LUA method 'update'.
C:/Users/User/Documents/My Games/FarmingSimulator2025/mods/FS25_ContractBoost/scripts/Settings.lua:36: attempt to index nil with table
2024-12-26 19:29   Info: FS25_ContractBoost:SETTINGS :: settingChanged table: 0x0000020c4723c088 14 customMaxPerType
2024-12-26 19:29 Error: Running LUA method 'update'.
C:/Users/User/Documents/My Games/FarmingSimulator2025/mods/FS25_ContractBoost/scripts/Settings.lua:36: attempt to index nil with table
2024-12-26 19:29   Info: FS25_ContractBoost:SETTINGS :: settingChanged table: 0x0000020c4723c088 15 customMaxPerType
2024-12-26 19:29 Error: Running LUA method 'update'.
C:/Users/User/Documents/My Games/FarmingSimulator2025/mods/FS25_ContractBoost/scripts/Settings.lua:36: attempt to index nil with table

**Comment by GMNGjoy** [comment]
@FirenzeIT `1.3.0.49` was not an official release, as it was not included on the releases page. 

Please do not pull any versions that are not announced as a release - this is an active code repository and individual commits are not intended to be "releases" unless published on the [Releases](https://github.com/GMNGjoy/FS25_ContractBoost/releases) page.

**Comment by FirenzeIT** [comment]
> @FirenzeIT `1.3.0.49` was not an official release, as it was not included on the releases page.
> 
> Please do not pull any versions that are not announced as a release - this is an active code repository and individual commits are not intended to be "releases" unless published on the [Releases](https://github.com/GMNGjoy/FS25_ContractBoost/releases) page.

Ok, so i don't and can't report the errors here?? o.O

**Comment by FirenzeIT** [comment]
Hi, warning with v1.0.5.0 (2024-12-27 12:47   Warning: FS25_ContractBoost:LOAD Could not read ContractBoost settings from either g_currentMission or savegameSettingsFile) it's normal if not have any savegame??

**Comment by GMNGjoy** [comment]
> > @FirenzeIT `1.3.0.49` was not an official release, as it was not included on the releases page.
> > Please do not pull any versions that are not announced as a release - this is an active code repository and individual commits are not intended to be "releases" unless published on the [Releases](https://github.com/GMNGjoy/FS25_ContractBoost/releases) page.
> 
> Ok, so i don't and can't report the errors here?? o.O
> 
> A warning with v1.0.5.0: (2024-12-27 12:47 Warning: FS25_ContractBoost:LOAD Could not read ContractBoost settings from either g_currentMission or savegameSettingsFile) it's normal if not have any savegame??

THAT is not what I said. I welcome bug reports or feature request through GitHub - but _ONLY ON RELEASED VERSIONS_. Unless it's a version that was released via the [Releases](https://github.com/GMNGjoy/FS25_ContractBoost/releases) page, it's not intended for use. Your [post above](https://github.com/GMNGjoy/FS25_ContractBoost/issues/34#issuecomment-2563017579) was on a version that could only have been obtained by manually downloading the repository itself, which I don't want, or support. I ONLY support _RELEASED VERSIONS._ If I tried to support every commit version, I'd go crazy.

The way that GitHub works is that the "code" itself (the repository) is a place to store "versioned" code. When I'm developing, I'm constantly saving "commits" which are points in time where I think something is ready, but I'm still testing. I don't want, and don't expect _anyone_ to be manually pulling that code, zipping it up for themselves, and then playing with it. Unless it's on the Releases page, please don't touch it.

I only post to the Releases page when something is tested and working as I expect it to. _Not before_.

> A warning with v1.0.5.0: (2024-12-27 12:47 Warning: FS25_ContractBoost:LOAD Could not read ContractBoost settings from either g_currentMission or savegameSettingsFile) it's normal if not have any savegame??

that is normal, but I'll de-classify it as a warning. This state happens either when you're activating the mod for the first time, or you're upgrading from a older version (pre-`1.0.2.1`), or you're connecting to a dedicated server. It just means that the default "location" where the mod gets it's settings from in SP (the savegameSettingsFile, `savegame##/FS25_ContractBoost.xml`) is not there yet, and it has to figure out what to do next.

**Comment by FirenzeIT** [comment]
Hi @GMNGjoy, ok. However, the silage bales selling, it's fix them soon??

**Comment by GMNGjoy** [comment]
@FirenzeIT I'm looking into it, and don't have an ETA or even know if it's fixable. I'm doing what I can, please be patient and wait for a release. If you'd like to discuss more, or ask other questions - feel free to join my [discord](http://gmngjoy.com/discord); This is an issue tracker, not a discussion forum.

That said, it was a mistake to close this yesterday since it's not fixed yet, I must have clicked the wrong button.

**Comment by GMNGjoy** [comment]
fixed in [1.0.6.0](https://github.com/GMNGjoy/FS25_ContractBoost/releases/tag/1.0.6.0)

**Comment by FirenzeIT** [comment]
Hi, very nice works, now when selling the wrapped bales from contracts, it's possible without have any issues/errors/bugs, nice! But i effort a probably other bug: contract bale wrap it's blocked at 99%, without have any bales not wrapped available in the field, it's possible have this chance? Map: Riverbend Springs; Mods: not others than ContractBoost.
